bedrock/wordpress/models.py

```diff
@@ -0,0 +1,113 @@
+import operator
+from django.conf import settings
+from django.db import models, transaction
+from django.db.models import Q
+from django.db.utils import DatabaseError
+from django.utils.html import strip_tags
+
+from django_extensions.db.fields.json import JSONField
+from jinja2 import Markup
+
+from bedrock.wordpress.api import get_posts_data
+
+
+class BlogPostQuerySet(models.QuerySet):
+    def filter_by_blog(self, blog_slug):
+        return self.filter(wp_blog_slug=blog_slug)
+
+    def filter_by_tags(self, *tags):
+        tag_qs = [Q(tags__contains='"{}"'.format(t)) for t in tags]
+        return self.filter(reduce(operator.or_, tag_qs, Q()))
+
+
+class BlogPostManager(models.Manager):
+    def get_queryset(self):
+        return BlogPostQuerySet(self.model, using=self._db)
+
+    def filter_by_blog(self, blog_slug):
+        return self.get_queryset().filter_by_blog(blog_slug)
+
+    def filter_by_tags(self, *tags):
+        return self.get_queryset().filter_by_tags(*tags)
+
+    def update_posts(self, data, database='default'):
+        with transaction.atomic(using=database):
+            count = 0
+            posts = data['posts']
+            self.filter_by_blog(data['wp_blog_slug']).delete()
+            for post in posts:
+                try:
+                    self.create(
+                        wp_id=post['id'],
+                        wp_blog_slug=data['wp_blog_slug'],
+                        date=post['date_gmt'],
+                        modified=post['modified_gmt'],
+                        title=post['title']['rendered'],
+                        excerpt=post['excerpt']['rendered'],
+                        link=post['link'],
+                        featured_media=post['featured_media'],
+                        tags=post['tags'],
+                    )
+                    count += 1
+                except DatabaseError:
```

Comment:
I'm not sure what kind, but the idea was to try to prevent the scenario you mention. I thought by catching some strange error with a single post we could store the ones we could instead of canceling the whole thing. But, perhaps I should let the exception bubble up since that might cancel the transaction and thus not delete everything?